src/components/MapView/index.tsx

```diff
@@ -357,14 +367,37 @@ function MapView({ classes }: MapViewProps) {
     // let users know if the layers selected are not possible to view together.
     if (
       selectedLayerDates.length === 0 &&
-      selectedLayersWithDateSupport.length !== 0
+      selectedLayersWithDateSupport.length !== 0 &&
+      selectedDate
     ) {
+      const layerToRemove = selectedLayers[selectedLayers.length - 2];
+      const layerToKeep = selectedLayers[selectedLayers.length - 1];
+
       dispatch(
         addNotification({
-          message: 'No dates overlap with the selected layers.',
+          message: `No dates overlap with the selected layers. Removing previous layer: ${layerToRemove.id}`,
           type: 'warning',
         }),
       );
+
+      // Remove layer from url.
+      const urlLayerKey = getUrlKey(layerToRemove);
+      const updatedUrl = removeLayerFromUrl(urlLayerKey, layerToRemove.id);
+      if (updatedUrl === '') {
+        removeKeyFromUrl(urlLayerKey);
+      } else {
+        updateHistory(urlLayerKey, updatedUrl);
+      }
+      dispatch(removeLayer(layerToRemove));
```

Comment:
This part feels overly complicated, I see that this logic is used in another place already but I believe that it can be simplified. Again, this can be addressed later but I believe that we can:
1. better separate the responsibilities between holding layers and managing the URL
2. make `removeLayerFromUrl` a bit smarter and move some of the logic in there